src/lib.rs

```diff
@@ -79,6 +262,45 @@ impl Environment {
             self.vars
         }
     }
+
+    pub fn with_var<T: Into<Var>>(mut self, v: T) -> Environment {
+        let var = v.into();
+        self.vars.push((var.key, var.value));
+
+        self
+    }
+
+    pub fn var<T: ToString>(self, key: T) -> Var {
+        let key = OsString::from(key.to_string());
+
+        Var {
+            key: key.clone(),
+            value: if self.inherit {
+                if let Some(v) = ::std::env::vars_os()
+                    .chain(self.vars.clone())
+                    .filter(|&(ref k, _)| k == &key)
```

Comment:
1. Should the inherit logic be moved to the constructors?
2. Would it make sense to not walk all of `var_os` but if its not found in `self.vars`, to call `env::var`?